File: backend/Models/Employee.cs
using System.ComponentModel.DataAnnotations.Schema;

namespace Employee_Management_System.Models
{
	public class Employee
	{
		// Todo: change Id to GUID later
		[DatabaseGenerated(DatabaseGeneratedOption.Identity)]
		public int Id { get; set; }
		public string FirstName { get; set; } = string.Empty;
		public string LastName { get; set; } = string.Empty;
		public string Email { get; set; } = string.Empty;
		public string PhoneNumber { get; set; } = string.Empty;
		public string Position { get; set; } = string.Empty;
		public string Department { get; set; } = string.Empty;

	}
}

File: frontend/src/app/employee/employee.ts
export class Employee {
    constructor(
      public firstName: string | null,
      public lastName: string | null,
      public email: string | null,
      public phoneNumber: string | null,
      public position: string | null,
      public department: string | null,
      public id?: number,
    ) {}
  }
  